Packs/CortexAttackSurfaceManagement/Scripts/InferWhetherServiceIsDev/InferWhetherServiceIsDev.py

```diff
@@ -28,31 +28,39 @@ def _canonicalize_string(in_str: str) -> str:
     return in_str.lower().strip(' \t"\'')
 
 
-def get_indicators_from_key_value_pairs(observed_key_value_pairs: list, is_indicator_match: Callable) -> list:
+def get_indicators_from_list(observed_list: list, is_indicator_match: Callable, comparison_type: str) -> list:
     """
     Returns list of matches based on criteria.
 
     Args:
-        observed_key_value_pairs (List[str]): list of tags to process.
+        observed_list (List[str]): list of tags to process.
         is_indicator_match (callable): what function to call depending on dev or prod checking.
+        comparison_type (str): if comparing list of dictionaries or a list of strings.
 
     Returns:
         list: list of matches based on exact/partial dev criteria.
     """
     indicators = []
-    for kv_pair in observed_key_value_pairs:
-        if not isinstance(kv_pair, Mapping):
-            demisto.info(f"Ignoring item because it is not a mapping: {kv_pair}")
-        else:
-            if "key" not in kv_pair or "value" not in kv_pair:
-                demisto.info(f"Ignoring item because it lacks the keys 'key' and/or 'value': {sorted(kv_pair.keys())}")
+    for list_entry in observed_list:
+        if comparison_type == "dictionary":
+            if not isinstance(list_entry, Mapping):
+                demisto.info(f"Ignoring item because it is not a mapping: {list_entry}")
             else:
-                key = _canonicalize_string(kv_pair.get("key", ""))
-                value = _canonicalize_string(kv_pair.get("value", ""))
-
-                if (("env" in key) or (key in ("stage", "function", "lifecycle", "usage", "tier"))) and is_indicator_match(value):
-                    indicators.append(kv_pair)
-
+                if "key" not in list_entry or "value" not in list_entry:
+                    demisto.info(f"Ignoring item because it lacks the keys 'key' and/or 'value': {sorted(list_entry.keys())}")
+                else:
+                    key = _canonicalize_string(list_entry.get("key", ""))
+                    value = _canonicalize_string(list_entry.get("value", ""))
+
+                    if (("env" in key) or (key in ("stage", "function", "lifecycle", "usage", "tier"))) and \
+                       is_indicator_match(value):
+                        indicators.append(list_entry)
+        elif comparison_type == "string":
```

Comment:
i might pull this logic out into a separate function to handle `list[str]` inputs, while preserving this one for `list[dict]` inputs, but that just be a matter of preference since functionally they're equivalent

Reply:
[PR feedback](https://github.com/demisto/content/pull/30601/commits/cc59bddcbcfddffbc2e558a72a565654b0bfa5d1)